### PR TITLE
Let operations on MINPVV modify initial values set by MINPV/MINPORV

### PR DESCRIPTION
**Pull request by blattms**
Previously any modifications were ignored if MINPV or MINPORV was specified in the model. 
Hence in this example:
```
MINPV 200
EQUALS
  MINPVV ...
```
the simulator would have used 200 as the minimum pore volume of each cell and MINPVV would have had no effect. This is in contrast to something like
```
-- No MINPV/MINPORV in GRID section
MINPVV 100*200
EQUALS
  MINPVV ...
```
where equals would actually change the initial values.

With this patch the behavior is now the same for both cases.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

File: opm/input/eclipse/EclipseState/Grid/FieldData.hpp [hunk 1]
```diff
@@ -190,6 +190,26 @@ namespace Opm::Fieldprops {
                       value::status::valid_default);
         }
 
+        void default_assign_global(const std::vector<T>& src)
+        {
+            if (!global_data) {
+                throw std::invalid_argument {
+                    "Cannot call default_assign_global on keyword with local storag"
```
Comment (akva2) [inline opm/input/eclipse/EclipseState/Grid/FieldData.hpp:197]: storage

Comment (blattms) [reply opm/input/eclipse/EclipseState/Grid/FieldData.hpp:197]: Are you a bot?

**Comment by blattms** [comment]
I think this correct but would like to make some checks first. Hence draft.

Jenkins should fail for test MINPV300

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this please

**Comment by tskille** [comment]
This PR has been tested on a full field model in Equinor and results are excellent. Looking forward to see this going into master repo.

File: opm/input/eclipse/EclipseState/Grid/FieldData.hpp [hunk 2]
```diff
@@ -190,6 +191,41 @@ namespace Opm::Fieldprops {
...
+            if (!global_data) {
+                throw std::invalid_argument {
+                    "Cannot call update_local_from_gloabl on keyword with local storage"
+                };
+            }
+            std::size_t i{};
+            for(auto current = this->data.begin(); current != this->data.end(); ++current, ++i)
+            {
+                this->data[i] = (*global_data)[local_to_global(i)];
+                this->value_status[i] = (*global_value_status)[local_to_global(i)];
```
Comment (bska) [inline opm/input/eclipse/EclipseState/Grid/FieldData.hpp:225]: You're calling `local_to_global` twice here, and you're not using `current` (which corresponds to `this->data[i]`).  Is that intentional?

File: tests/parser/EclipseGridTests.cpp [hunk 3]
```diff
@@ -956,6 +956,48 @@ static Opm::Deck createMinpvvEqualsCPDeck() {
     return parser.parseString( deckData) ;
 }
 
+static Opm::Deck createMinpvEqualsMinpvvCPDeck() {
+    const char* deckData =
+        "RUNSPEC\n"
+        "\n"
```
Comment (bska) [inline tests/parser/EclipseGridTests.cpp:962]: Is there a reason for not using raw string literals in the new test here?

File: opm/input/eclipse/EclipseState/Grid/FieldData.hpp [hunk 4]
```diff
@@ -190,6 +191,41 @@ namespace Opm::Fieldprops {
...
+                    ", expected: " + std::to_string(this->dataSize())
+                };
+            }
+
+            std::copy(src.begin(), src.end(), this->global_data->begin());
+            std::fill(this->global_value_status->begin(), this->global_value_status->end(),
+                      value::status::valid_default);
+        }
+
+        void update_local_from_global(std::function<std::size_t(std::size_t)> local_to_global)
```
Comment (bska) [inline opm/input/eclipse/EclipseState/Grid/FieldData.hpp:214]: Very strictly speaking you don't need `std::function<>` here.  We're in a header and could just as well make the function a template instead.

**Review by bska — COMMENTED** [review]
Looks good for the most part, although there is probably some potential performance gain lingering in a few of the interface and implementation choices here.

I do wish we'd switch to using raw string literals for new tests though.

File: opm/input/eclipse/EclipseState/Grid/FieldData.hpp (hunk 4 above)
Comment (blattms) [reply opm/input/eclipse/EclipseState/Grid/FieldData.hpp:214]: I could also have passed EclipseGrid with the complete type, but I really did not want to pull in that header here. This class was initially internal to opm-common.

I doubt that there are any real performance issues due to the way it is now and might compile faster (famous last words without any proofs :grimacing:)

File: opm/input/eclipse/EclipseState/Grid/FieldData.hpp (hunk 2 above)
Comment (blattms) [reply opm/input/eclipse/EclipseState/Grid/FieldData.hpp]: Good catch, fixed.

File: tests/parser/EclipseGridTests.cpp (hunk 3 above)
Comment (blattms) [reply tests/parser/EclipseGridTests.cpp]: Nope, just simple copy&paste and my usual laziness (to look up how they and linebreak replacement in emacs works).

Changed,

**Review by blattms — COMMENTED** [review]
I hopefully addressed all relevant comments.

**Comment by blattms** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me and I'll merge into master.  It's not clear to me what the semantics of the (hypothetical) reverse case
```
EQUALS
  'MINPVV' 5E3 ... /
/
-- ...
MINPV
  300 /
```
would be or if we transparently handle that now, but we should nevertheless defer such handling until we have a case that uses it.

**Comment by blattms** [comment]
> Thanks a lot for the updates. This looks good to me and I'll merge into master. It's not clear to me what the semantics of the (hypothetical) reverse case
> 
> ```
> EQUALS
>   'MINPVV' 5E3 ... /
> /
> -- ...
> MINPV
>   300 /
> ```
> 
> would be or if we transparently handle that now, but we should nevertheless defer such handling until we have a case that uses it.

No. We will assume a wrong order as we always process MINPORV/MINPV before the MINPVV no matter how they are actually ordered.

**Pull request merged**